### PR TITLE
Tweak: Improve performance

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: src/shared/utils.js [hunk 1]
```diff
@@ -1,92 +1,77 @@
...
  *
- * @return {String|null} - Winner value or `null`.
+ * @return { rows: Array, cols: Array, diags: Array };
  */
-export function getWinner( board ) {
-	// Check rows & cols.
-	for ( let i = 0; i < board.length; i++ ) {
-		const row = board[ i ];
+export function extractBoardParts( board ) {
+	const size = parseInt( Math.sqrt( board.length ) );
```
Comment (rotem227) [inline src/shared/utils.js]: Why do you need the parseInt ?

File: src/shared/utils.js [hunk 2]
```diff
@@ -1,92 +1,77 @@
...
+		// Calculate rows / cols
+		for ( let j = 0; j < size; j++ ) {
+			rows[ index ].push( board[ i + j ] );
+			cols[ index ].push( board[ i / 3 + j * size ] );
+		}
 
-	// Check RTL diagonal.
-	const diagRTL = getDiagonal( board, 'rtl' );
+		// Calculate diagonals.
+		const iteration = i / 3;
```
Comment (rotem227) [inline src/shared/utils.js]: Move the iteration variable above the rows/col loop and use it in line 41.

File: src/shared/utils.js (hunk 1 above)
Comment (StyleShit) [reply src/shared/utils.js]: Removed

File: src/shared/utils.js (hunk 2 above)
Comment (StyleShit) [reply src/shared/utils.js]: Done

**Pull request merged**